models/classes/session/Business/Service/SessionCookieService.php

```diff
@@ -15,67 +15,103 @@
  * along with this program; if not, write to the Free Software
  * Foundation, Inc., 51 Franklin Street, Fifth Floor, Boston, MA  02110-1301, USA.
  *
- * Copyright (c) 2020 (original work) Open Assessment Technologies SA;
+ * Copyright (c) 2020-2022 (original work) Open Assessment Technologies SA;
  *
  * @author Sergei Mikhailov <sergei.mikhailov@taotesting.com>
  */
 
-declare(strict_types=1);
+declare (strict_types=1);
 
 namespace oat\tao\model\session\Business\Service;
 
 use common_http_Request as Request;
 use oat\tao\model\service\InjectionAwareService;
 use oat\tao\model\session\Business\Contract\SessionCookieAttributesFactoryInterface;
 use oat\tao\model\session\Business\Contract\SessionCookieServiceInterface;
+use oat\tao\model\session\Business\Domain\SessionCookieAttribute;
 use tao_helpers_Uri as UriHelper;
 
 class SessionCookieService extends InjectionAwareService implements SessionCookieServiceInterface
 {
     /** @var SessionCookieAttributesFactoryInterface */
     private $sessionCookieAttributesFactory;
+    private $attributes = [];
+    private $sessionCookieParams = [];
 
     public function __construct(SessionCookieAttributesFactoryInterface $sessionCookieAttributesFactory)
     {
         parent::__construct();
-
         $this->sessionCookieAttributesFactory = $sessionCookieAttributesFactory;
+        //it will brake the unit test logic
+        //   $this->init();
     }
 
-    public function initializeSessionCookie(): void
+    private function init(): void
     {
+        $iterator = $this->sessionCookieAttributesFactory->create()->getIterator();
+        while ($iterator->valid()) {
+            $this->attributes[] = $iterator->current();
+            $iterator->next();
+        }
+
         $sessionParams = session_get_cookie_params();
-        $cookieDomain  = UriHelper::isValidAsCookieDomain(ROOT_URL)
+        $cookieDomain = UriHelper::isValidAsCookieDomain(ROOT_URL)
             ? UriHelper::getDomain(ROOT_URL)
             : $sessionParams['domain'];
-        $isSecureFlag  = Request::isHttps();
-        $sessionCookieAttributes = $this->sessionCookieAttributesFactory->create();
+        $isSecureFlag = Request::isHttps();
+        if (isset($sessionParams['lifetime'])) {
+            $this->attributes[] = new SessionCookieAttribute('lifetime', $sessionParams['lifetime']);
+        }
+
+        $this->attributes[] = new SessionCookieAttribute('domain', $cookieDomain);
+        $this->attributes[] = new SessionCookieAttribute('secure', $isSecureFlag);
+        $this->attributes[] = new SessionCookieAttribute('httponly', true);
```

Comment:
```suggestion
        if (isset($sessionParams['lifetime'])) {
            $attributes[] = new SessionCookieAttribute('lifetime', $sessionParams['lifetime']);
        }

        $attributes[] = new SessionCookieAttribute('domain', $cookieDomain);
        $attributes[] = new SessionCookieAttribute('secure', $isSecureFlag);
        $attributes[] = new SessionCookieAttribute('httponly', true);
```
reason of that mentioned https://github.com/oat-sa/tao-core/pull/3712/files#r1057772958